### ISSUE TITLE
7x7 Matrix grafisch darstellen

### ISSUE THREAD

**Issue by Birdy2404** [S-Man42/GCWizard#474]
Geocache GC46A1J - Agent P. ermittelt #1: Manuelle Kryptographie - https://coord.info/GC46A1J

Der wäre auch was für die grafische Darstellung ähnlich zisterzienser und Braille.
Lammers ist der Erfinder von Lotto

**Comment by t-m-z** [comment]
will do it similar to maya, babylon...

**Comment by S-Man42** [comment]
Wait, please not that quick :) Is the 7x7 matrix something special for the cache or something real existing? I think we should avoid creating a tool for a code, someone created for one single mystery. We should take care of the tool list... Something which is only used once, should not be part of app. Otherwise we won't get happy anymore. Afterwards, someone comes and creates 8x12 matrix or whatever... I don't think that makes any sense.

I know a 5x7 matrix, which is commonly used on modern hi-fi devices for example. But a 7x7 matrix?

**Comment by t-m-z** [comment]
hae a look at Geocache GC46A1J - Agent P. ermittelt #1: Manuelle Kryptographie - https://coord.info/GC46A1J
7x7 is a Lotto-Field  ;-)

**Comment by S-Man42** [comment]
ooookaaaaay.... You want to create a Lotto... schein...? :D

**Comment by t-m-z** [comment]
;-D  like the segment Display- Tool

Sent from MailDroid

-----Original Message-----

**Comment by S-Man42** [comment]
OK, maybe with an option:
German Lotto 6 from 49 (7x7)
German Toto 6 from 45 (6x7 + 3)
Eurojackpot 5 from 50 (5 x 10)
...

When implementing, please have a look at my Babylonian Implementation, which made the Painter code much shorter, especially when we are talking about 50 same elements ;) It is only necessary to write the element code only once...

One difference to other display tools (btw: We should rename the base engine, we are at least here far away from any kind of segment displays :D ): All others will be painted only in another color if they are activated. Here it would be nice, if there get an X additionally. What do you think?

**Comment by t-m-z** [comment]
regarding
- the different kinds/options: okay
- coding: sure
- cross: yes, would fit due to the context, but maybe with restricted visibility. The squares are used as a fot matrix. Hence using crosses could be hard to identify the encoded symbol. I have to check this

Sent from MailDroid

-----Original Message-----

**Comment by S-Man42** [comment]
I quickly build a prototype... I recommend, NOT using the segment display engine. It's much too overpowered for it (since you just want to fill or check the field, but not need to store any segment configurations - just a simple Tap On and Tap Off...). It is more like the Sudoku board (which already offers the grid and paints the numbers into the grid) with an underlying boolean grid as data structure. 

Using the Sudoku Engine needs to be changed to a variable col/row count (instead fixed 9x9): As parameter I recommend colCount and totalCount, so you can calculation the necessary rows like:

rowCount = (totalCount / colCount).ceil() 

(for example Toto's last row is not complete)

**Comment by t-m-z** [comment]
I am not quite sure if this would help solving https://coord.info/GC46A1J
I thought about an input-field to read the numbers and fill the table or tap on the different fields and get their numbers/get the segment configuration.

**Comment by S-Man42** [comment]
Hm, don't know...Maybe additionally. I would do both inputs (tap lottery dingens directly and enter numbers as input field)

**Comment by S-Man42** [comment]
Is this still in the pipeline for 2.0.0 or should we pospone it?

**Comment by t-m-z** [comment]
please postpone

Sent from MailDroid

-----Original Message-----

**Comment by S-Man42** [comment]
OK, had some time to think about this. I really like to implement this, but will not have much time. So I'd like to describe my thoughts:

**First View:**

One dropdown incl.
3x3
4x4
...
10x10
Lotto (10x5)
Toto (...)
Chess Board
Polybios
...
Whatever

Below the drawn grid with enumerated boxes.

Below one row with colored squared:
Black, White, Red, Yellow, Blue, Green.
These can be selected. Selected color will be used to mark the box (transparent overlay over entire field)

**Additional dropdown entry "Custom".** Now it becomes interesting:
Tap on it opens a second view (Customizing, like the Point/Line Editors in Open Map)

Switch: Tappable Boxes <-> Tappable Grid Lines (sometimes you need the boxes, sometimes you want to mark the lines, like in UP DOWN LEFT RIGHT)

2 Integerspinners:
X/Y
(draws x+1 lines when mode == boxes)

Section: Labels:
Dropdown:
No Labels
Label Boxes (e.g. like the Lotto grid)
   --> Works only in Mode == boxes
   --> Show another Dropdown for direction; several Cases
         -- Start Top Left, go rowwise
         -- Start Top Left, go columnwise
         -- Start Top Right, go rowwise, ....
   -- > Switch for option if

```
                          123    >>>>
                          456    >>>>
                          789    >>>>
```

or

```
                           123  >>>>>
                           654  <<<<<               <-- switch direction each line or row
                           789  >>>>>
```
   --> Values: Use Range Input like VarCoords or FormulaRangeValues like: "1-100#2" (which is 1,3,5,7,9,11)
        --> If number values less then available boxes: Leave boxes empty. If more, than ignore too much values.
   --> If no numbers, use free text field (e.g. for ABCDEFGHIKLMN...)
Label Rows/Columns (more like Chess Board)
   --> When Mode == boxes: Label rows/columns between line
   --> When Mode == grid: Label lines
   --> Input fields for column and row (input like boxes, see above)

Option to store configured boards.

**Back in First View:**

"Grid" Mode: Each section of line is tappable. Should be drawn a bit bigger than the grid. Optional draw the grid lines at top an left a bit longer. If tapped on those extensions, it will mark the entire line

"Box" Mode: Each Box is tappable. Grid should be draw a bit bigger as well... Can mark the entire row or column as well.

**Comment by S-Man42** [comment]
This would create a highly configurable grid system for many different needs. I guess that would be extremely helpful!

**Comment by S-Man42** [comment]
First step into the right direction: https://github.com/S-Man42/GCWizard/tree/grid

**Comment by S-Man42** [comment]
Logic completely works;

TODO:
- Configuration widgets
- Default configurations
- Code refactoring

**Comment by S-Man42** [comment]
#848
